### PR TITLE
Fix issue 100

### LINKED ISSUE #100: Refactor `Enrgy_sub`, move all `Enrgy_` functions
In STEMMUS_SCOPE.m, all `Enrgy_` functions are called using the script `Enrgy_sub`:

https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/d2fe0a671ab26430f086e1c8ee8272f4db6fa060/src/STEMMUS_SCOPE.m#L751

However, for model maintainability we want to remove all use of globals and running scripts.

Additionally, as all `Enrgy_` functions seem to tackle one specific part of the model, these should be moved to a separate Matlab package with a appropriate name.

<hr>

To do:
- [ ] Move all `Enrgy_` files to their own package, with an appropriate name
- [ ] Convert `Enrgy_sub` into a function. Use structures to reduce the amount of input and output variables.
- [ ] Add documentation to all functions
- [ ] Make the required changes in the main file.
- [ ] Create a pull request for the changes

### PR DESCRIPTION
**Pull request by SarahAlidoost**
### Description

closes #100. I tested this branch with three site data. The results are the same as those generated by the main branch. 

### Checklist
- [x] Add a reference to related issues.
- [x] @mentions of the person or team responsible for reviewing proposed changes.
- [x] This pull request has a descriptive title.
- [x] Code is written according to the [guidelines](http://cnl.sogang.ac.kr/cnlab/lectures/programming/matlab/Richard_Johnson-MatlabStyle2_book.pdf).
- [x] The checks by [MISS_HIT style checker and linter](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/CONTRIBUTING.md#follow-matlab-style-guidelines), below the pull request, are successful (green).
- [x] Documentation is available.
- [x] Add changes to the [changelog file](CHANGELOG.md) under section `Unreleased`.
- [x] Model runs successfully.
- [x] Ask a meinatainer to re-generate exe file if matlab codes are changed. About how to create an exe file, see [exe readme](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/run_model_on_snellius/exe/README.md).

### CODE REVIEW

File: src/+energy/calculateBoundaryConditions.m [hunk 1]
```diff
@@ -0,0 +1,43 @@
+function [RHS, EnergyMatrices] = calculateBoundaryConditions(BoundaryCondition, EnergyMatrices, HBoundaryFlux, ForcingData, ...
+                                                             SoilVariables, Precip, EVAP, Delt_t, r_a_SOIL, Rn_SOIL, RHS, L, KT)
+    %{
+        Determine the boundary condition for solving the energy equation.
```
Comment (BSchilperoort) [inline src/+energy/calculateBoundaryConditions.m:4]: This function is missing a reference. Is one available (e.g. the STEMMUS technical manual) or is it undocumented?

File: src/+energy/calculateEnergyFluxes.m [hunk 2]
```diff
@@ -0,0 +1,10 @@
+function [QET, QEB] = calculateEnergyFluxes(SAVE, TT)
+    %{
+        Calculate the energy fluxes on the boundary nodes.
```
Comment (BSchilperoort) [inline src/+energy/calculateEnergyFluxes.m:3]: Here a documentation reference is missing as well

**Review by BSchilperoort — APPROVED** [review]
:partying_face: 
Just some small comments.

Again I assume you tested for any changes in model result.

File: src/+energy/calculateBoundaryConditions.m (hunk 1 above)
Comment (SarahAlidoost) [reply src/+energy/calculateBoundaryConditions.m]: Looking at issue #100, I assume the reference is STEMMUS Technical Notes. Added.

File: src/+energy/calculateEnergyFluxes.m (hunk 2 above)
Comment (SarahAlidoost) [reply src/+energy/calculateEnergyFluxes.m]: Looking at issue #100, I assume the reference is STEMMUS Technical Notes. Added.

**Comment by SarahAlidoost** [comment]
> 🥳 Just some small comments.

Thanks for reviewing :smile: 

> Again I assume you tested for any changes in model result.

yes, I mentioned it [here](https://github.com/EcoExtreML/STEMMUS_SCOPE/pull/197#issue-1921510825).

**Pull request merged**